### PR TITLE
feat(ci): Add Vale for writing linter in markdown files

### PR DESCRIPTION
**Pull request by Kayx23**
Integrated [Vale](https://vale.sh) / [Vale Action](https://github.com/errata-ai/vale-action) into CI to check spellings and grammar in markdown files.
* action triggered on PR and is run on markdown files (*md) 
   * review results for changed markdown files in the PR are showed in [Findings](https://github.com/Kayx23/trakydeng.ca/actions/runs/4496489490/jobs/7911216246) while the rest are in Filtered Findings
* Vale injects annotations into files in **Files changed** in a PR ([example](https://github.com/Kayx23/trakydeng.ca/pull/38/files))
* APISIX specific wording allowlist & denylist are found in `./.vale/Vocab/APISIX` (should continue improve this list)
* levels of review messages: `INFO`, `WARNING`, `ERROR`; Github Action fails on `ERROR` level

### CODE REVIEW

**Review by juzhiyuan — APPROVED** [review]
LGTM

**Comment by An-DJ** [comment]
LGTM

**Review by An-DJ — APPROVED** [review]

File: .github/workflows/vale-grammar-linter.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+on: [pull_request]
+
+jobs:
+  vale-linter:
+    name: runner / vale
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Get changed files in the PR
+        run: |
```
Comment (Kayx23) [inline .github/workflows/vale-grammar-linter.yml:11]: `Get changed files in the PR` fetches the latest changes from the base branch of the pull request and compares them to the current branch. It then saves the names of the changed files to a txt file.

File: .github/workflows/vale-grammar-linter.yml [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+          git fetch origin ${{ github.event.pull_request.base.ref }}:refs/remotes/origin/base
+          git diff --name-only --diff-filter=d $(git merge-base HEAD origin/base)..HEAD > changed_files_in_PR.txt
```
Comment (Kayx23) [inline .github/workflows/vale-grammar-linter.yml:13]: `Get changed files in the PR` fetches the latest changes from the base branch of the pull request and compares them to the current branch. It then saves the names of the changed files to a txt file.

**Pull request merged**